core/arch/arm/mm/core_mmu.c

```diff
@@ -837,12 +843,14 @@ static void init_mem_map(struct tee_mmap_region *memory_map, size_t num_elems)
 	assert(va >= CFG_TEE_RAM_START);
 	assert(end <= CFG_TEE_RAM_START + CFG_TEE_RAM_VA_SIZE);
 
-	add_pager_vaspace(memory_map, num_elems, va, end, &last);
+	add_pager_vaspace(memory_map, num_elems, va, end, &last,
+			  &next_pgdir_va);
 
```

Comment:
I think here, if pager is enabled, it is mandatory that the pager vmem space is fully reserved to pager. Hence, we should maybe add `va = ROUNDDOWN(va, CORE_MMU_PGDIR_SIZE);`, the same way we now have the pager vmem end provided through `next_pgdir_va`.

Maybe more simple to have `add_pager_vaspace(memory_map, num_elems, &va, &end, &last);` and the function to update `va` and `end` content if the virtual memory area reserved to TEE_RAM management must be enlarged due to pager constraint.